examples/auto_ml_checkpoint.py

```diff
@@ -0,0 +1,133 @@
+import shutil
+import time
+
+import numpy as np
+
+from neuraxle.base import BaseStep, NonFittableMixin
+from neuraxle.checkpoints import DefaultCheckpoint
+from neuraxle.hyperparams.distributions import RandInt
+from neuraxle.hyperparams.space import HyperparameterSpace
+from neuraxle.metaopt.random import RandomSearch
+from neuraxle.pipeline import ResumablePipeline, DEFAULT_CACHE_FOLDER, Pipeline
+
+EXPECTED_OUTPUTS = np.array(range(10, 20))
+
+DATA_INPUTS = np.array(range(10))
+
+HYPERPARAMETER_SPACE = HyperparameterSpace({
+    'multiplication_2__hp_mul': RandInt(1, 3),
+    'multiplication_3__hp_mul': RandInt(1, 3),
+    'multiplication_4__hp_mul': RandInt(1, 3),
+    'multiplication_5__hp_mul': RandInt(1, 3),
+    'multiplication_6__hp_mul': RandInt(1, 3),
+    'multiplication_7__hp_mul': RandInt(1, 3),
+    'multiplication_8__hp_mul': RandInt(1, 3),
+    'multiplication_9__hp_mul': RandInt(1, 3),
+    'multiplication_10__hp_mul': RandInt(1, 3),
+    'multiplication_11__hp_mul': RandInt(1, 3)
+})
+
+
+class Multiplication(NonFittableMixin, BaseStep):
+    def __init__(self, sleep_time=0.1, hyperparams=None, hyperparams_space=None):
```

Comment:
Since this class is used in the notebook and this test, it would be worth coding issue #99 already and also adding a `Sleeper()` step in the `misc` package file. You could add sleepers at some points in the pipeline in your example, interlaced with multiplication steps.